### PR TITLE
fix: NPE when udf metrics enabled

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

fixes: https://github.com/confluentinc/ksql/issues/5890

Fixes an NPE thrown when the second instance of a UDAF is initialized. The NPE was being thrown due to a bug in the code that failed to initialise the sensor fields in the `UdafAggregateFunction` class if the sensor already existed.

The commit also refactors the code used to add function invocation metrics such that all three function types: UDF, UDTF and UDAF, use the same function to register metrics.

Unit tests added to ensure this single code path doesn't throw an NPE or return null on the subsequent invocation.

Also added `ksql.udf.collect.metrics` metric to the server config file, so users are more likely to be able to find it and try it out.

Moved UDTF functions out of the `ksql-udf` group and into `ksql-udtf` group. Having them in the former was another bug.

### Testing done 

Added unit tests plus manual validation that metric names didn't change with new code, (except intentional move of UDTF metrics into own group).

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/FunctionMetrics.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      final String sensorName,
+      final String groupName,
+      final String functionDescription
+  ) {
+    final Sensor existing = metrics.getSensor(sensorName);
+    if (existing != null) {
+      return existing;
+    }
+
+    final Sensor newSensor = metrics.sensor(sensorName);
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/function/FunctionMetrics.java:79]: there's a race here that is possible here where two calls for the same `metrics.getSensor(sensorName)` don't find the sensor and they both end up registering the extra metrics. I don't know of a good way around it...

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/UdtfLoader.java [hunk 2]
```diff
@@ -67,7 +68,8 @@ public void loadUdtfFromClass(
     }
     final String functionName = udtfDescriptionAnnotation.name();
     final String sensorName = "ksql-udtf-" + functionName;
-    FunctionLoaderUtils.addSensor(sensorName, functionName, metrics);
+
+    FunctionMetrics.initInvocationSensor(metrics, sensorName, "ksql-udtf", functionName + " udtf");
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/function/UdtfLoader.java:72]: if anyone was relying on the old metrics, their alerts for udtfs will break with this change. probably ok as I'd be shocked if anyone was actually writing their own UDTFs _and_ using metrics on them at this point but something to call out 😉

Comment (big-andy-coates) [reply ksqldb-engine/src/main/java/io/confluent/ksql/function/UdtfLoader.java:72]: Yeah, I'm aware, but ... meh.  turning metrics on caused an NPE, so likely no one is using them, and better to change it now.

Comment (big-andy-coates) [reply ksqldb-engine/src/main/java/io/confluent/ksql/function/UdtfLoader.java:72]: As mentioned in the description, I see this as a bug fix.

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/FunctionMetrics.java (hunk 1 above)
Comment (big-andy-coates) [reply ksqldb-engine/src/main/java/io/confluent/ksql/function/FunctionMetrics.java]: Yeah I know, but I was just fixing the NPE and being lazy ;). Fixed it.

**Pull request merged**